hoprd/db/api/src/metadata.rs

```diff
@@ -0,0 +1,263 @@
+use async_trait::async_trait;
+use sea_orm::{ColumnTrait, EntityTrait, QueryFilter};
+use serde::{Deserialize, Serialize};
+
+use crate::{db::HoprdDb, errors::Result};
+
+/// Represents an alias
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct AliasEntry {
+    pub peer_id: String,
+    pub alias: String,
+}
+
+impl AliasEntry {
+    pub fn new(peer_id: String, alias: String) -> Self {
+        Self { peer_id, alias }
+    }
+}
+
+/// Defines DB API for accessing HOPR settings (mainly aliases for now)
+#[async_trait]
+pub trait HoprdDbMetadataOperations {
+    /// Retrieve the alias for a given PeerID
+    async fn resolve_alias(&self, alias: String) -> Result<Option<String>>;
+
+    /// Retrive all aliases
+    async fn get_aliases(&self) -> Result<Vec<AliasEntry>>;
+
+    /// Create new key pair value in the db
+    async fn set_alias(&self, peer: String, alias: String) -> Result<()>;
+
+    /// Update aliases
+    async fn set_aliases(&self, aliases: Vec<AliasEntry>) -> Result<()>;
+
+    /// Delete alias
+    async fn delete_alias(&self, alias: String) -> Result<()>;
+
+    /// Delete all aliases
+    async fn clear_aliases(&self) -> Result<()>;
+}
+
+#[async_trait]
+impl HoprdDbMetadataOperations for HoprdDb {
+    async fn resolve_alias(&self, alias: String) -> Result<Option<String>> {
+        let row = hoprd_db_entity::aliases::Entity::find()
+            .filter(hoprd_db_entity::aliases::Column::Alias.eq(alias))
+            .one(&self.metadata)
+            .await?;
+
+        Ok(row.map(|model| model.peer_id))
+    }
+
+    async fn get_aliases(&self) -> Result<Vec<AliasEntry>> {
+        let rows = hoprd_db_entity::aliases::Entity::find().all(&self.metadata).await?;
+
+        let aliases = rows
+            .iter()
+            .map(|row| AliasEntry {
+                peer_id: row.peer_id.clone(),
+                alias: row.alias.clone(),
+            })
```

Comment:
Create a `From<AliasEntry> for hoprd_db_entity::aliases::Model` to avoid these manual remappings.